hal_st/stm32fxxx/SpiMasterStm.cpp

```diff
@@ -95,61 +117,63 @@ namespace hal
     void SpiMasterStm::HandleInterrupt()
     {
         uint32_t status = peripheralSpi[spiInstance]->SR;
-#if !defined(STM32WBA)
+#ifdef SPI_FLAG_RXNE
         if ((status & SPI_FLAG_RXNE) != 0)
 #else
-        if ((status & SPI_FLAG_RXWNE) != 0)
+        if ((status & SPI_IT_RXP) != 0)
 #endif
         {
             if (dummyToReceive != 0)
             {
-#if !defined(STM32WBA)
-                (void)peripheralSpi[spiInstance]->DR;
-#else
+#ifdef SPI_RXDR_RXDR
                 (void)peripheralSpi[spiInstance]->RXDR;
+#else
+                (void)peripheralSpi[spiInstance]->DR;
 #endif
                 --dummyToReceive;
             }
             else if (receiving)
             {
-#if !defined(STM32WBA)
-                receiveData.front() = peripheralSpi[spiInstance]->DR;
-#else
+#ifdef SPI_RXDR_RXDR
                 receiveData.front() = peripheralSpi[spiInstance]->RXDR;
+#else
+                receiveData.front() = peripheralSpi[spiInstance]->DR;
 #endif
                 receiveData.pop_front();
             }
 
             receiving &= !receiveData.empty();
 
             if (dummyToReceive == 0 && !receiving)
-#if !defined(STM32WBA)
-                peripheralSpi[spiInstance]->CR2 &= ~SPI_IT_RXNE;
+#ifdef SPI_IER_RXPIE
+                peripheralSpi[spiInstance]->IER &= ~SPI_IT_RXP;
 #else
-                peripheralSpi[spiInstance]->CR2 &= ~SPI_IT_RXP;
+                peripheralSpi[spiInstance]->CR2 &= ~SPI_IT_RXNE;
```

Comment:
Is SPI_IT_RXNE valid for the CR2 register?

Reply:
Good catch: I've changed it wrongly. Thanks.

Reply:
Yeap. This definition is valid for CR2 in all other families in hal_st.

![image](https://github.com/philips-software/amp-hal-st/assets/162123637/b415f33e-9e5f-478c-8e52-7aba68039680)